src/util/svg-text-wrapper.js

```diff
@@ -0,0 +1,99 @@
+const TextWrapper = require('./text-wrapper');
+
+/**
+ * Measure text by using a hidden SVG attached to the DOM.
+ * For use with TextWrapper.
+ */
+class SVGMeasurementProvider {
+    constructor () {
+        this._svgRoot = null;
+    }
+
+    /**
+     * Detach the hidden SVG element from the DOM and forget all references to it and its children.
+     */
+    dispose () {
+        if (this._svgRoot) {
+            this._svgRoot.parent.removeChild(this._svgRoot);
+            this._svgRoot = null;
+            this._svgText = null;
+        }
+    }
+
+    /**
+     * Called by the TextWrapper before a batch of zero or more calls to measureText().
+     */
+    beginMeasurementSession () {
+        if (!this._svgRoot) {
+            this._init();
+        }
+    }
+
+    /**
+     * Called by the TextWrapper after a batch of zero or more calls to measureText().
+     */
+    endMeasurementSession () {
+        this._svgText.textContent = '';
+    }
+
+    /**
+     * Measure a whole string as one unit.
+     * @param {string} text - the text to measure.
+     * @returns {number} - the length of the string.
+     */
+    measureText (text) {
+        this._svgText.textContent = text;
+        return this._svgText.getComputedTextLength();
+    }
+
+    /**
+     * Create a simple SVG containing a text node, hide it, and attach it to the DOM. The text node will be used to
+     * collect text measurements. The SVG must be attached to the DOM: otherwise measurements will generally be zero.
+     * @private
+     */
+    _init () {
+        const svgNamespace = 'http://www.w3.org/2000/svg';
+
+        const svgRoot = document.createElementNS(svgNamespace, 'svg');
+        const svgGroup = document.createElementNS(svgNamespace, 'g');
+        const svgText = document.createElementNS(svgNamespace, 'text');
+
+        // Normalize text attributes to match what the svg-renderer does.
+        // @TODO This code should be shared with the svg-renderer.
```

Comment:
@cwillisf so for this PR, do you think I should:
(a) Leave this the way it is
(b) Extract the text formatting part of the svg renderer into a utility and use it both here and in the svg renderer.
(c) Something else?